Add ENGAGE-HF App and its module dependencies

eldcn opened the pull request:
# *Add Engage app and its module dependencies*

## :recycle: Current situation & Problem
- Moved engage app and modules
- Spezi modules are currently copied, and will be replaced once we publish
- Adapted workflows
- Adapted project structure. All Engage specific modules are grouped under `engagehf-modules` and spezi ones under `spezi`:

```
.
└── ENGAGE-HF-Android/
    ├── app
    ├── engagehf-modules
    └── spezi
```
## :gear: Release Notes
*Add a bullet point list summary of the feature and possible migration guides if this is a breaking change so this section can be added to the release notes.*
*Include code snippets that provide examples of the feature implemented or links to the documentation if it appends or changes the public interface.*


## :books: Documentation
*Please ensure that you properly document any additions in conformance to [Spezi Documentation Guide](https://github.com/StanfordSpezi/.github/blob/main/DOCUMENTATIONGUIDE.md).*
*You can use this section to describe your solution, but we encourage contributors to document your reasoning and changes using in-line documentation.* 


## :white_check_mark: Testing
*Please ensure that the PR meets the testing requirements set by CodeCov and that new functionality is appropriately tested.*
*This section describes important information about the tests and why some elements might not be testable.*


### Code of Conduct & Contributing Guidelines
By creating and submitting this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

pauljohanneskraft reviewed (approved):
Thank you for these changes! Looks great - I would have personally kept the name `modules` for the engage-specific modules (since this is now an engage-specific repo), but I'm also fine with renaming the directory!

eldcn commented:
> Thank you for these changes! Looks great - I would have personally kept the name `modules` for the engage-specific modules (since this is now an engage-specific repo), but I'm also fine with renaming the directory!

@pauljohanneskraft I renamed the namespaces also from edu.stanford.spezi.modules to edu.stanford.bdh.engagehf.modules, hence also renamed the parent directory. Once we publish spezi modules, we can move those top level then. This is just temporary to distinguish between spezi and engage specific modules
